src/borrow_check/type_check.md

```diff
@@ -8,3 +8,59 @@ this type-check, we also uncover the region constraints that apply to
 the program.
 
 TODO -- elaborate further? Maybe? :)
+
+## User types
+
+At the start of MIR type-check, we replace all regions in the body with new unconstrained regions.
+This would cause us to accept the following program however:
+```rust
+fn foo<'a>(x: &'a u32) {
+    let y: &'static u32 = x;
+}
+```
+By erasing the lifetimes in the type of `y` we no longer know that it is supposed to be `'static`,
+ignoring the intentions of the user.
+
+To deal with this we remember all places where the user explicitly mentioned a type during
+hir typeck as [`CanonicalUserTypeAnnotations`][annot].
```

Comment:
Can you replace "hir typeck" with "HIR type-check" or "`hir` type-check"? (And likewise "hir" -> "`hir`" or "HIR".) Using the shorthand is likely to confuse new contributors, especially since in the dev guide we usually spell stuff out more.